File: leet_1103_candies.py
class Solution:
    def distributeCandies(self, candies: int, num_people: int) -> List[int]:
        res = [0] * num_people
        idx = 0
        c = 1
        while candies:
            n = min(c, candies)
            candies -= n
            res[idx] += n
            idx = (idx + 1) % num_people
            c += 1
        return res

File: leet_1103_candies.cc
class Solution {
public:
    vector<int> distributeCandies(int candies, int num_people)
    {
        vector<int> res(num_people, 0);
        int c = 0;
        int idx = 0;
        int n;
        while (++c && candies)
        {
            n = c < candies ? c : candies;
            candies -= n;
            res[idx] += n;
            idx++;
            idx %= num_people;
        }
        return res;

    }
};
